### ISSUE TITLE
Can Custom Select component be used without livewire?

### ISSUE THREAD

**Issue by luilliarcec** [rawilk/laravel-form-components#34]
Reviewing the code a little, I found a fragment that would imply that it is.

```blade
{{-- custom-select.blade.php --}}

@unless ($attributes->hasStartsWith('wire:model'))
    <input type="hidden" x-bind:value="JSON.stringify(value)" name="{{ $name }}">
@endif
```

However, when testing it in the traditional way, leaving Livewire aside, I noticed that both for the single and multiple select the component was broken and it did not recover the values ​​of the old request either.

I managed to fix this by replacing the above code with the following.

```blade
@unless ($attributes->hasStartsWith('wire:model'))
    @if ($multiple)
        <template x-for="item in value">
            <input type="hidden" x-bind:value="item" name="{{ $name }}[]">
        </template>
    @else
        <input type="hidden" x-bind:value="value" name="{{ $name }}">
    @endif
@endunless
```

However when making that change, it was normal for something to break so I received the following error when working with multiselect. It should be noted that the single select works wonderfully with this change.

```
TypeError
htmlspecialchars(): Argument #1 ($string) must be of type string, array given (View: ..\laravel-form-components\resources\views\components\inputs\custom-select.blade.php)
```

But hey, I already knew where the thing was going, so I investigated some more code and in the php class of the custom select modify the following function.

```php
// CustomSelect.php

public function selectedKeyToJS(): mixed
{
    if (is_null($this->selectedKey)) {
        return "''";
    }

    return is_string($this->selectedKey)
        ? "'{$this->selectedKey}'"
        : json_encode($this->selectedKey);
}
```

That corrected the error. But a new bug came up now in JS.

![image](https://user-images.githubusercontent.com/27895611/132856700-6159865d-307a-4753-8cdf-3ad6771ea058.png)

I would appreciate any help, or if you have tried to do it before without success, I would appreciate knowing it to desist from continuing.

**Comment by rawilk** [comment]
I haven't been able to replicate these errors on my end, however I did implement your change of creating a hidden input for each selected value for multiple selects.

I didn't change anything else, as it seems to be working just fine and when I submitted my test form, I received the values in the controller.

I'm not sure what version of alpine you're using, but if it's not the latest version it may fix any issues you have by updating to it.

**Comment by rawilk** [comment]
Also, in the future please make use of the issue templates as they will have you enter some more context that will be helpful in debugging issues like this.

I'm going to close this for now since I can't replicate it, but feel free to re-open in the future with some more information if the issue still persists for you.

**Comment by rawilk** [comment]
Actually, I think I've figured out what's causing the problem and I'm working on a fix for it. I'll update this issue once I find out more.

**Comment by rawilk** [comment]
Should be resolved with new `6.0.0` version being released today

**Comment by luilliarcec** [comment]
Thanks so much
